lark/exceptions.py

```diff
@@ -107,18 +107,21 @@ def match_examples(self, parse_fn: 'Callable[[str], Tree]',
                     parse_fn(malformed)
                 except UnexpectedInput as ut:
                     if ut.state == self.state:
-                        if use_accepts and hasattr(self, 'accepts') and hasattr(ut, 'accepts') and ut.accepts != self.accepts:
+                        # mypy doesn't currently support hasattr() with literal attribute names
+                        # https://github.com/python/mypy/issues/1424
+                        if use_accepts and hasattr(self, 'accepts') and hasattr(ut, 'accepts') and ut.accepts != self.accepts:  # type: ignore[attr-defined]
                             logger.debug("Different accepts with same state[%d]: %s != %s at example [%s][%s]" %
-                                         (self.state, self.accepts, ut.accepts, i, j))
+                                         (self.state, self.accepts, ut.accepts, i, j))  # type: ignore[attr-defined]
                             continue
                         try:
-                            if ut.token == self.token:  # Try exact match first
+                            # Try exact match first
+                            if ut.token == self.token:  # type: ignore[attr-defined]
                                 logger.debug("Exact Match at example [%s][%s]" % (i, j))
                                 return label
 
                             if token_type_match_fallback:
                                 # Fallback to token types match
-                                if (ut.token.type == self.token.type) and not candidate[-1]:
+                                if (ut.token.type == self.token.type) and not candidate[-1]:  # type: ignore[attr-defined]
                                     logger.debug("Token Type Fallback at example [%s][%s]" % (i, j))
                                     candidate = label, True
```

Comment:
Since this and the previous block are both attempting to achieve the same goal, it is odd that they use different strategies (checking first vs letting it fail). It would probably be "better" pick a single strategy and use it for both, but I wanted to keep this PR to just addressing the mypy errors instead of changing code.